homeassistant/components/homekit_controller/__init__.py

```diff
@@ -149,12 +150,15 @@ def get_characteristic_types(self):
         raise NotImplementedError
 
 
-def setup(hass, config):
+async def async_setup(hass, config):
     """Set up for Homekit devices."""
     # pylint: disable=import-error
     import homekit
     from homekit.controller.ip_implementation import IpPairing
 
+    map_storage = hass.data[ENTITY_MAP] = EntityMapStorage(hass)
+    await map_storage.async_initialize()
+
     hass.data[CONTROLLER] = controller = homekit.Controller()
 
     for hkid, pairing_data in load_old_pairings(hass).items():
```

Comment:
This does I/O, which isn't allowed in a coroutine. Please add it as an executor job.

Reply:
Oops. We already made this async in config flow and hadn't originally split this PR apart from config flow. Will fix this one now.